wagtail/images/management/commands/wagtail_update_image_renditions.py

```diff
@@ -14,43 +19,55 @@ def add_arguments(self, parser):
+            help="Operate in x size chunks (default: %(default)s)",
+        )
 
     def handle(self, *args, **options):
-        renditions = get_image_model().get_rendition_model().objects.all()
-        if len(renditions) == 0:
+        Rendition = get_image_model().get_rendition_model()
+
+        renditions = Rendition.objects.all()
+
+        purge_only = options["purge_only"]
+
+        if not renditions.exists():
             self.stdout.write("No image renditions found.")
             return
 
-        success_count = 0
-        if options["purge_only"]:
-            for rendition in renditions:
-                try:
-                    rendition_image = rendition.image
-                    rendition.delete()
-                    success_count = success_count + 1
-                except Exception:  # noqa: BLE001
-                    self.stderr.write(
-                        f"Could not purge rendition for {rendition_image.title}"
-                    )
+        rendition_ids = list(renditions.values_list("id", flat=True))
+
+        if purge_only:
             self.stdout.write(
-                self.style.SUCCESS(
-                    f"Successfully purged {success_count} image rendition(s)"
-                )
+                self.style.HTTP_INFO(f"Purging {len(rendition_ids)} rendition(s)")
             )
         else:
-            for rendition in renditions:
-                try:
+            self.stdout.write(
+                self.style.HTTP_INFO(f"Regenerating {len(rendition_ids)} rendition(s)")
+            )
+
+        for rendition in (
+            renditions.filter(id__in=rendition_ids)
+            .select_related("image")
+            .iterator(chunk_size=options["chunk_size"])
+        ):
+            try:
+                with transaction.atomic():
                     rendition_filter = rendition.filter
                     rendition_image = rendition.image
+
+                    # Delete the existing rendition
                     rendition.delete()
-                    rendition_image.get_rendition(rendition_filter)
-                    success_count = success_count + 1
-                except Exception:  # noqa: BLE001
-                    self.stderr.write(
-                        f"Could not regenerate rendition for {rendition_image.title}"
-                    )
-            self.stdout.write(
-                self.style.SUCCESS(
-                    f"Successfully regenerated {success_count} image rendition(s)"
+
+                    if not purge_only:
+                        # Create a new one
+                        rendition_image.get_rendition(rendition_filter)
+            except:  # noqa:E722
+                logger.exception("Error operating on rendition %d", rendition.id)
+                self.stderr.write(
+                    self.style.ERROR(f"Failed to operate on rendition {rendition.id}")
                 )
-            )
+
+        self.stdout.write(self.style.SUCCESS("Success"))
```

Comment:
nitpick:

```suggestion
        self.stdout.write(self.style.SUCCESS("Done"))
```

or "Finished" may be better. if the process fails for all renditions, "success" at the end it not what it says on the tin :)

Also, `wagtail.images.tests.test_management_commands.TestUpdateImageRenditions` are failing because of the changed messages